core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -51,16 +52,23 @@
 
     private final Pipeline.Factory factory = new Pipeline.Factory();
     private final Map<String, Processor.Factory> processorFactories;
+    private Boolean newIngestDateFormat;
```

Comment:
This field should be `'volatile` now, since it gets updated by a different thread than it is read from.
Maybe also change the type to a primitive boolean?